### PR TITLE
Added possibility to pass parameters to a mapping profile

### PR DESCRIPTION
**Pull request by martynasramonas**

### CODE REVIEW

File: Source/Mapping.AutoMapper/MappingProfile.cs [hunk 1]
```diff
@@ -28,18 +28,30 @@ public class Mapper : IMapper<TSource, TDestination>
...
+
+            public virtual TDestination Map(TSource source, params (string parameterName, object parameterValue)[] parameters)
+            {
+                return mapper.Map<TSource, TDestination>(source, opt =>
+                {
+                    if (parameters != null)
+                    {
+                        foreach ((string parameterName, object parameterValue) parameter in parameters)
+                        {
+                            if (!string.IsNullOrWhiteSpace(parameter.parameterName) && parameter.parameterValue != null)
```
Comment (johannesvaltonen) [inline Source/Mapping.AutoMapper/MappingProfile.cs]: Should we be able to pass nulls, too?

File: Source/Mapping.AutoMapper/TwoWayMappingProfile.cs [hunk 2]
```diff
@@ -28,6 +28,23 @@ public virtual TSource Map(TDestination source)
             {
                 return mapper.Map<TDestination, TSource>(source);
             }
+
+            public virtual TSource Map(TDestination source, params (string parameterName, object parameterValue)[] parameters)
```
Comment (johannesvaltonen) [inline Source/Mapping.AutoMapper/TwoWayMappingProfile.cs]: Can we call the Map method on the base class to reduce code duplication? Or move the method contents to a protected static methods and call it from both Map methods?

Comment (johannesvaltonen) [reply Source/Mapping.AutoMapper/TwoWayMappingProfile.cs]: What I actually think we should do, is to deprecate the TwoWayMapper altogether. It's part of the old implementation and not really used in any of our newer projects anymore. We could think about adding "Obsolete" attribute to all constructs related to two-way mappng.

File: Source/Mapping.AutoMapper/MappingProfile.cs (hunk 1 above)
Comment (martynasramonas) [reply Source/Mapping.AutoMapper/MappingProfile.cs]: I agree. It's over-checking

File: Source/Mapping.AutoMapper/TwoWayMappingProfile.cs (hunk 2 above)
Comment (martynasramonas) [reply Source/Mapping.AutoMapper/TwoWayMappingProfile.cs]: I have removed the code duplication. 
I agree about TwoWayMapper. It usage really reduce code readability. I will add obsolete attribute.

File: Source/Mapping.AutoMapper/ITwoWayMappingProfile.cs [hunk 3]
```diff
@@ -1,7 +1,9 @@
-﻿using AutoMapper;
+﻿using System;
+using AutoMapper;
 
 namespace Affecto.Mapping.AutoMapper
 {
+    [Obsolete]
```
Comment (johannesvaltonen) [inline Source/Mapping.AutoMapper/ITwoWayMappingProfile.cs]: Should we add a comment, too? [ObsoleteAttribute("Two-way mapping is deprecated and it is going to be removed in a future major version.", false)]

File: Source/Mapping.AutoMapper/TwoWayMappingProfile.cs (hunk 2 above)
Comment (johannesvaltonen) [reply Source/Mapping.AutoMapper/TwoWayMappingProfile.cs]: Should we then even add this new method to the two-way mapper? Probably not if we are going to remove those classes.

File: Source/Mapping/Mapping.csproj [hunk 4]
```diff
@@ -9,9 +9,9 @@
     <Company>Affecto</Company>
     <PackageProjectUrl>https://github.com/affecto/dotnet-Mapping</PackageProjectUrl>
     <PackageTags>affecto map mapping</PackageTags>
-    <PackageReleaseNotes>Converted to .NET Standard 2.0</PackageReleaseNotes>
+    <PackageReleaseNotes>Added possibility to pass parameters to a mapping profile</PackageReleaseNotes>
```
Comment (johannesvaltonen) [inline Source/Mapping/Mapping.csproj]: "Added possibility to pass parameters to a mapping profile. Deprecated two-way-mapping."

File: Source/Mapping.AutoMapper/ITwoWayMappingProfile.cs (hunk 3 above)
Comment (johannesvaltonen) [reply Source/Mapping.AutoMapper/ITwoWayMappingProfile.cs]: Sorry, I meant [Obsolete("Two-way mapping is deprecated and it is going to be removed in a future major version.", false)]

File: Source/Mapping.AutoMapper/TwoWayMappingProfile.cs (hunk 2 above)
Comment (martynasramonas) [reply Source/Mapping.AutoMapper/TwoWayMappingProfile.cs]: Ok. Removed

**Pull request merged**